marklogic-client-api/src/main/java/com/marklogic/client/datamovement/HostAvailabilityListener.java

```diff
@@ -129,15 +132,22 @@ public HostAvailabilityListener withSuspendTimeForHostUnavailable(Duration
    * @return this instance (for method chaining)
    */
   public HostAvailabilityListener withMinHosts(int numHosts) {
-    if (numHosts <= 0) throw new IllegalArgumentException("numHosts must be > 0");
-    int numConfigHosts = moveMgr.readForestConfig().getPreferredHosts().length;
-    if (numHosts > numConfigHosts) throw new IllegalArgumentException
-      ("numHosts must be less than or equal to the number of hosts in the cluster");
+    if (moveMgr.getConnectionPolicy() == DatabaseClient.ConnectionPolicy.PRIMARY_HOST) {
+      if (numHosts != 0) {
+        throw new IllegalArgumentException("numHosts must be 1 when using only the primary host for the connection");
```

Comment:
Shouldn't the condition be  if (numHosts != 1) ?

Reply:
Yes, good catch.